repair/repair.cc

```diff
@@ -2058,6 +2082,12 @@ static future<> do_decommission_removenode_with_repair(seastar::sharded<database
                 }
                 neighbors_set.erase(myip);
                 neighbors_set.erase(leaving_node);
+                // Remove nodes in ignore_nodes
+                if (ops) {
+                    for (const auto& node : ops->ignore_nodes) {
+                        neighbors_set.erase(node);
+                    }
+                }
```

Comment:
Maybe we should abort the repair. Aren't our quorum calculations invalidated by the node removal?

What happens if the removal happens after we pass this point?

Reply:
Ah, I see this is for the removenode itself.

Maybe check the parameters for consistency? If doing a decommission, ignore_nodes must be empty, and vice versa.

Reply:
We have:
```
2135 future<> decommission_with_repair(seastar::sharded<database>& db, seastar::sharded<netw::messaging_service>& ms, locator::token_metadata_ptr tmptr) {
2136     return do_decommission_removenode_with_repair(db, ms, std::move(tmptr), utils::fb_utilities::get_broadcast_address(), {});
2137 }
```

It is guaranteed the node_ops is empty, thus node_ops->ignore_nodes is empty, for decommission.